ibis/expr/types/core.py

```diff
@@ -48,7 +48,18 @@ def __repr__(self) -> str:
             )
             return output
         else:
-            return repr(result)
+            try:
+                from rich.console import Console
```

Comment:
also i would generally put these imports at module level (and protect them) so calling this w/o the import won't fail repeatedly by trying to import (which can be expensive as well)

Reply:
A failed import is even cheaper

Reply:
sure but again this can be a hidden performance issue in a large graph (imagine it fails every time then it's extremely expensive)

Reply:
no maybe we have a special case but the issue we have seen is basically this

a try/except in some operations will format an exception as part of the except operation (but maybe just is flow control)

since repr is invoked there (and could be very expensive) graph construction in some edge cases can be very large cost

it doesn't happen often but it can be very hard to do

so i think strategically repr should be very cheap and other display modes should be intentional

Reply:
True, but is that different from the existing code? I'm not saying that what you're describing doesn't happen, only that this code has no effect on what you're describing.